ants/registration/registration.py

```diff
@@ -1770,40 +1778,74 @@ def label_image_registration(fixed_label_images,
         if verbose:
             print("\n\nComputing deformable transform using images.\n")
 
-        do_quick = False
-        do_repro = False
-
-        if "Quick" in type_of_deformable_transform:
-            do_quick = True
-        elif "Repro" in type_of_deformable_transform:
-            do_repro = True
-            random_seed = str(1)
-
-        intensity_metric_parameter = None
+        intensity_metric = "CC"
+        intensity_metric_parameter = 2
+        syn_shrink_factors = "8x4x2x1"
+        syn_smoothing_sigmas = "3x2x1x0vox"
+        syn_convergence = "[100x70x50x20,1e-6,10]"
         spline_distance = 26
-        if "[" in type_of_deformable_transform and "]" in type_of_deformable_transform:
-            subtype_of_deformable_transform = type_of_deformable_transform.split("[")[1].split("]")[0]
-            if not ('bo' in subtype_of_deformable_transform or 'so' in subtype_of_deformable_transform):
-                raise ValueError("Only 'so' or 'bo' transforms are available.")
-            if "," in subtype_of_deformable_transform:
-                subtype_of_deformable_transform_args = subtype_of_deformable_transform.split(",")
-                subtype_of_deformable_transform = subtype_of_deformable_transform_args[0]
-                intensity_metric_parameter = subtype_of_deformable_transform_args[1]
-                if len(subtype_of_deformable_transform_args) > 2:
-                    spline_distance = subtype_of_deformable_transform_args[2]
+        gradient_step = 0.1
+        syn_transform = "SyN"
 
         syn_stage = list() 
 
-        intensity_metric = None
-        if fixed_intensity_images is not None and len(fixed_intensity_images) > 0:
+        if isinstance(type_of_deformable_transform, list):
+
+            if (len(type_of_deformable_transform) != 7 or
+                not isinstance(type_of_deformable_transform[0], str) or
+                not isinstance(type_of_deformable_transform[1], float) or
+                not isinstance(type_of_deformable_transform[2], str) or
+                not isinstance(type_of_deformable_transform[3], int) or 
+                not isinstance(type_of_deformable_transform[4], tuple) or 
+                not isinstance(type_of_deformable_transform[5], tuple) or 
+                not isinstance(type_of_deformable_transform[6], tuple)):
+                raise ValueError("Incorrect specification for type_of_deformable_transform.")            
```

Comment:
People are going to open issues and just paste this message. Ideally the error would explain exactly what's wrong, or at least specify what the expected input should be

Reply:
Added a reference to the help menu.